apps/src/lab2/views/components/editor/CodeEditor.tsx

```diff
@@ -127,6 +128,33 @@ const CodeEditor: React.FunctionComponent<CodeEditorProps> = ({
     editorEditableCompartment,
   ]);
 
+  // Sets aria-label on the input div once code mirror loads
+  useEffect(() => {
+    const setAriaLabel = () => {
+      const cmContentDiv = editorRef.current?.querySelector('.cm-content');
+      if (cmContentDiv) {
+        cmContentDiv.setAttribute('aria-label', i18n.codeEditor());
+      }
+    };
+
+    // Set the aria-label initially
+    setAriaLabel();
+
+    // Ensure we set aria-label if/when code mirror updates
```

Comment:
We can update the editor by changing the start code or by changing it from read only to editable and vice versa. I'm not sure if that would remove the aria label though?

Reply:
I think it only would if it had some sort of dynamic reloading/refreshing! I just am not very familiar with CodeMirror generally😅

Reply:
Though I suppose if the CodeEditor reloads, it'll call that useEffect on the new load anyway! I'll remove that part